src/react/utils.js

```diff
@@ -814,8 +813,6 @@ export function getValueFromFunctionCall(
     } else {
       throw error;
     }
-  } finally {
-    invariant(createdObjects === realm.createdObjects, "realm.createdObjects was not correctly restored");
```

Comment:
Let's keep this invariant, it's been very useful in finding early evaluation errors. Rather, we should capture the effects of the function call (using `evaluateForEffects`) and apply them if the createdObjects has been correctly restored. We ran into countless issues in the past that this guarded against, so I don't think we should regress on this if possible.